### PR TITLE
Feat: add support for subenvironments

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of the effort to add support for sub environments #536.

In the PR I implemented the create operation.
A unit test was added.

### CODE REVIEW

File: env0/resource_environment.go [hunk 1]
```diff
@@ -348,6 +348,29 @@ func setEnvironmentSchema(d *schema.ResourceData, environment client.Environment
 		d.Set("approve_plan_automatically", !*environment.RequiresApproval)
 	}
 
+	if environment.LatestDeploymentLog.WorkflowFile != nil && len(environment.LatestDeploymentLog.WorkflowFile.Environments) > 0 {
```
Comment (TomerHeber) [inline env0/resource_environment.go:351]: This is for setting sub environments ids.
(This data is only available in the response).

File: env0/resource_environment.go [hunk 2]
```diff
@@ -698,6 +721,23 @@ func getCreatePayload(d *schema.ResourceData, apiClient client.ApiClientInterfac
...
+
+	subEnvironments, err := getSubEnvironments(d)
+	if err != nil {
+		return client.EnvironmentCreate{}, diag.Errorf("failed to extract subenvrionments from resourcedata: %v", err)
+	}
+
+	if len(subEnvironments) > 0 {
+		deployPayload.SubEnvironments = make(map[string]client.SubEnvironment)
+
+		for _, subEnvironment := range subEnvironments {
```
Comment (TomerHeber) [inline env0/resource_environment.go:733]: Sets the sub environments in the create request.

File: env0/resource_environment_test.go [hunk 3]
```diff
@@ -1532,3 +1532,123 @@ func TestUnitEnvironmentWithoutTemplateResource(t *testing.T) {
 	})
 
 }
+
+func TestUnitEnvironmentWithoutSubEnvironment(t *testing.T) {
```
Comment (TomerHeber) [inline env0/resource_environment_test.go:1536]: The unit test will be expanded after the rest is implemented.

Comment (TomerHeber) [reply env0/resource_environment_test.go:1536]: Update operation in unit test.

File: env0/resource_environment.go [hunk 4]
```diff
@@ -348,6 +348,29 @@ func setEnvironmentSchema(d *schema.ResourceData, environment client.Environment
 		d.Set("approve_plan_automatically", !*environment.RequiresApproval)
 	}
 
+	if environment.LatestDeploymentLog.WorkflowFile != nil && len(environment.LatestDeploymentLog.WorkflowFile.Environments) > 0 {
+		iSubEnvironments, ok := d.GetOk("sub_environment_configuration")
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/resource_environment.go:352]: correct. it's an array.

File: env0/resource_environment.go [hunk 5]
```diff
@@ -348,6 +348,29 @@ func setEnvironmentSchema(d *schema.ResourceData, environment client.Environment
...
+
+		if ok {
+			var newSubEnvironments []interface{}
+
+			for i, iSubEnvironment := range iSubEnvironments.([]interface{}) {
+				subEnviornment := iSubEnvironment.(map[string]interface{})
+
+				name := d.Get(fmt.Sprintf("sub_environment_configuration.%d.name", i)).(string)
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/resource_environment.go:360]: yes... this is a little confusing.

From an API perspective during create there's no "name"/"alias" fields. So I'm unsure what you mean.
We may need to meet to discuss this.

[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/resource_environment.go]: @eladmosh - modified.

**Review by eladmosh — APPROVED** [review]

**Pull request merged**